docs/USAGE-GUIDE.md

```diff
@@ -1018,6 +1018,12 @@ set the public keys found in the Certificate into **public_key_out**.
 s2n-tls includes support for resuming from cached SSL/TLS session, provided 
 the caller sets (and implements) three callback functions.
 
+### s2n\_config\_set\_session\_cache\_onoff
+
+**s2n_config_set_session_cache_onoff** enables the session cache. In order to use the session cache feature
+you will need to set a ticket key with **s2n_config_add_ticket_crypto_key** to encrypt session
```

Comment:
what will this function return if a ticket_crypto_key has not been set?

Reply:
Heh. It would return successful. It's just a function to indicate a user wants to use session cache, it doesn't check if a ticket key has been added.

Reply:
that seems weird. so a user could call this function and since it returns successfully they could go on to assume caching works. should this functions return an error if the key is not set? that would prevent an invalid state?

was there a reason for not returning an error?

Reply:
I think the general idea of the s2n on/off functions is to indicate which type of resumption the user wants to do, rather than checking that the user has the correct configurations to do that specific type of resumption. 🤷‍♀️ They will throw a S2N_ERR_NO_TICKET_ENCRYPT_DECRYPT_KEY error if you try to resume without adding a ticket key.